runtime/vm/FlushProcessWriteBuffers.cpp

```diff
@@ -37,50 +37,60 @@ extern "C" {
 void
 flushProcessWriteBuffers(J9JavaVM *vm)
 {
+	/* If this is called before the underlying data structures have been initialized,
+	 * assume that the VM is single-threaded and proceed without error.
+	 */
 #if defined(WIN32)
-	((VOID (WINAPI*)(void))vm->flushFunction)();
-#elif defined(LINUX) /* WIN32 */
-	omrthread_monitor_enter(vm->flushMutex);
-	void *addr = vm->exclusiveGuardPage.address;
-	UDATA pageSize = vm->exclusiveGuardPage.pageSize;
-	int mprotectrc = mprotect(addr, pageSize, PROT_READ | PROT_WRITE);
-	Assert_VM_true(0 == mprotectrc);
-	VM_AtomicSupport::add((UDATA*)addr, 1);
-	mprotectrc = mprotect(addr, pageSize, PROT_NONE);
-	Assert_VM_true(0 == mprotectrc);
-	omrthread_monitor_exit(vm->flushMutex);
-#else /* LINUX */
+	if (NULL != vm->flushFunction) {
+		((VOID (WINAPI*)(void))vm->flushFunction)();
+	}
```

Comment:
Should there be some runtime failure if the function does not exist? Similarly below for the mutex?

Reply:
No, the point is to be able to acquire exclusive before it has been initialized, on the assumption that we're single-threaded.  See comment above.

Reply:
If the initialize fails (can't allocate mutex/find function), the VM fails to start.

Reply:
Sorry missed the comment.